### DISCUSSION TITLE
How can I add scopes to an OpenIdApplication using a recipe?

### DISCUSSION THREAD

**Discussion by Rodien** [OrchardCMS/OrchardCore#15058] [Q&A] [answered]
Hi Guys,

Is it possible to add an OpenID scope, for example, 'openid' and 'email,' to an OpenID application using a recipe? If yes, how can I achieve this?

@hishamco

**Comment by hishamco** [comment]
@kevinchalet is our OpenId guru

**Reply by kevinchalet** [reply, marked as answer]
It's possible to control the scopes an application can request by adding a `ScopeEntries` node to your app recipe (not sure why it's called `ScopeEntries` instead of `Scopes` tho' 😅):

https://github.com/OrchardCMS/OrchardCore/blob/f190ddd31419e3e26e16df238d0c2031c1181312/src/OrchardCore.Modules/OrchardCore.OpenId/Recipes/OpenIdApplicationStepModel.cs#L15C7-L15C7

You'll also need a recipe to create the scopes themselves.

It seems https://docs.orchardcore.net/en/main/docs/reference/modules/OpenId/#client-openid-connect-apps-configuration is outdated as it doesn't mention that node.

**Reply by Rodien** [reply]
It works. Thank you!
